File: 2432-number-of-zero-filled-subarrays/number-of-zero-filled-subarrays.java
class Solution {
    public long zeroFilledSubarray(int[] nums) {
        long count = 0;
        long subArrayLength = 1;
        int index = 0;
        while (index < nums.length){
            while(index < nums.length && nums[index] == 0){
                count += subArrayLength;
                subArrayLength++;
                index++;
            }
            subArrayLength = 1;
            index++;
        }
        return count;
    }
}

File: 2432-number-of-zero-filled-subarrays/number-of-zero-filled-subarrays.cpp
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) 
    {
        long long count = 0;
        int i = 0;
        while(i < nums.size())
        {
            if(nums[i] == 0)
            {
                int length = 1;
                while(i < nums.size() && nums[i] == 0)
                {
                    count += length;
                    length++;
                    i++;
                }
            }
            else
            i++;
        }
        return count;
    }
};